easybuild/easyblocks/l/lammps.py

```diff
@@ -0,0 +1,360 @@
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+@author: Pavel Grochal (INUITS)
+@author: Kenneth Hoste (Ghent University)
+@author: Alan O'Cais
+
+Based on work of Alan O'Cais (https://github.com/ocaisa)
+https://github.com/easybuilders/easybuild-easyconfigs/blob/0c7fa07b9b7a855df6d14b971bd9eb1a25f51dd8/easybuild/easyconfigs/l/LAMMPS/LAMMPS-24Oct2018-intel-2018b.eb
+"""
+
+import os
+
+import easybuild.tools.toolchain as toolchain
+from easybuild.framework.easyconfig import CUSTOM, MANDATORY
+from easybuild.tools.build_log import EasyBuildError, print_warning, print_msg
+from easybuild.tools.config import build_option
+from easybuild.tools.environment import unset_env_vars
+from easybuild.tools.modules import get_software_root, get_software_version
+from easybuild.tools.run import run_cmd
+from easybuild.tools.systemtools import get_shared_lib_ext
+
+from easybuild.easyblocks.generic.cmakemake import CMakeMake
+
+KOKKOS_CPU_ARCH_LIST = [
+    'ARMv80',  # ARMv8.0 Compatible CPU
+    'ARMv81',  # ARMv8.1 Compatible CPU
+    'ARMv8-ThunderX',  # ARMv8 Cavium ThunderX CPU
+    'BGQ',  # IBM Blue Gene/Q CPUs
+    'Power8',  # IBM POWER8 CPUs
+    'Power9',  # IBM POWER9 CPUs
+    'SNB',  # Intel Sandy/Ivy Bridge CPUs
+    'HSW',  # Intel Haswell CPUs
+    'BDW',  # Intel Broadwell Xeon E-class CPUs
+    'SKX',  # Intel Sky Lake Xeon E-class HPC CPUs (AVX512)
+    'KNC',  # Intel Knights Corner Xeon Phi
+    'KNL',  # Intel Knights Landing Xeon Phi
+]
+
+KOKKOS_CPU_MAPPING = {
+    'sandybridge': 'SNB',
+    'ivybridge': 'SNB',
+    'haswell': 'HSW',
+    'broadwell': 'BDW',
+    'skylake_avx512': 'SKX',
+    'knights-landing': 'KNL',
+}
+
+
+KOKKOS_GPU_ARCH_TABLE = {
+    "3.0": "Kepler30",  # NVIDIA Kepler generation CC 3.0
+    "3.2": "Kepler32",  # NVIDIA Kepler generation CC 3.2
+    "3.5": "Kepler35",  # NVIDIA Kepler generation CC 3.5
+    "3.7": "Kepler37",  # NVIDIA Kepler generation CC 3.7
+    "5.0": "Maxwell50",  # NVIDIA Maxwell generation CC 5.0
+    "5.2": "Maxwell52",  # NVIDIA Maxwell generation CC 5.2
+    "5.3": "Maxwell53",  # NVIDIA Maxwell generation CC 5.3
+    "6.0": "Pascal60",  # NVIDIA Pascal generation CC 6.0
+    "6.1": "Pascal61",  # NVIDIA Pascal generation CC 6.1
+    "7.0": "Volta70",  # NVIDIA Volta generation CC 7.0
+    "7.2": "Volta72",  # NVIDIA Volta generation CC 7.2
+    "7.5": "Turing75",  # NVIDIA Turing generation CC 7.5
```

Comment:
nitpicking, but I prefer single quotes for string values that don't have spaces (more readable imho)